libcontainer/container_linux.go

```diff
@@ -1801,8 +1801,11 @@ func (c *linuxContainer) bootstrapData(cloneFlags uintptr, nsMaps map[configs.Na
 					Value: []byte(c.newgidmapPath),
 				})
 			}
-			// The following only applies if we are root.
-			if !c.config.Rootless {
+			gidMapCount := 0
+			for _, m := range c.config.GidMappings {
+				gidMapCount += m.Size
+			}
```

Comment:
This isn't quite right. It will miss if we have a single mapping of more than one group. Really, what we should do is have a helper function that does something like:

```
func requiresMappingTool(c *configs.Config) bool {
  gidMap := []configs.IDMap{
    {ContainerID: 0, HostID: os.Getegid(), Size: 1},
  }
  return !reflect.DeepEquals(c.GidMappings, gidMap)
}
```

Reply:
> It will miss if we have a single mapping of more than one group. 

Why? Does not `gidMapCount += m.Size` catch such case?

Reply:
It doesn't catch if the `HostID` is not `os.Getegid`.

Reply:
thanks for pointing out, updated